clients/da-vinci-client/src/main/java/com/linkedin/davinci/storage/StorageService.java

```diff
@@ -371,6 +376,42 @@ public synchronized AbstractStorageEngine openStore(
     return engine;
   }
 
+  public synchronized void checkWhetherStoragePartitionsShouldBeKeptOrNot(SafeHelixManager manager) {
+    if (manager == null) {
+      return;
+    }
+    for (AbstractStorageEngine storageEngine: getStorageEngineRepository().getAllLocalStorageEngines()) {
+      String storageEngineName = storageEngine.getStoreVersionName();
+      String storeName = Version.parseStoreFromKafkaTopicName(storageEngineName);
+      Set<Integer> storageEnginePartitionIds = storageEngine.getPartitionIds();
+      PropertyKey.Builder propertyKeyBuilder =
+          new PropertyKey.Builder(configLoader.getVeniceClusterConfig().getClusterName());
+      SafeHelixDataAccessor helixDataAccessor = manager.getHelixDataAccessor();
+      IdealState idealState = helixDataAccessor.getProperty(propertyKeyBuilder.idealStates(storeName));
+      String instanceHostName = manager.getInstanceName();
+
+      if (idealState != null) {
+        Set<Integer> idealStatePartitionIds = new HashSet<>();
+        Set<String> partitionSet = idealState.getPartitionSet();
+        partitionSet.stream().forEach(partitionDbName -> {
+          idealStatePartitionIds.add(RocksDBUtils.parsePartitionIdFromPartitionDbName(partitionDbName));
+        });
+
+        Map<String, Map<String, String>> mapFields = idealState.getRecord().getMapFields();
+        for (Map.Entry<String, Map<String, String>> entry: mapFields.entrySet()) {
```

Comment:
Instead of looping over every partition in `idealState`, can we instead loop over the partitions we have locally and validate it against the `mapFields`?

Since the partition Ids we have locally are ints, we have to convert them to the format that `mapFields` has them (Test_Store_Migration_Demo_v1_0).
I think this would be the format: `storeName + localPartitionId`

Reply:
(Just to confirm, this is completed).